### PR TITLE
Update Footer with 'fallback links' menu item props

### LINKED ISSUE #382: Update Footer with 'fallback links' menu item props
.. so if the API call fails for some reason (Drupal or Contentful failing being one), the footer can still point to useful information on a more stable page, namely:

`lite.sportrelief.com/legal` and 
`lite.sportrelief.com/legal#privacy-policy`  (updates still to be made to lite site)

### PR DESCRIPTION
**Pull request by LouiaR**
Fixes #382

### CODE REVIEW

File: src/components/Menu/Menu.js [hunk 1]
```diff
@@ -65,38 +65,55 @@ class Menu extends Component {
...
       );
     }
-
-    return <div />;
+    return (
+      <nav className="menu--footer">
+        <ul className="menu" id={`${type}-menu`}>
+          <div>
+            <li className="menu-item">
+              <a href={campaign === 'sportrelief' ? 'https://lite.sportrelief.com/terms-of-use' : 'https://lite.comicrelief.com/legal/'} rel="noopener noreferrer" target="_blank">Legal</a>
```
Comment (AndyEPhipps) [inline src/components/Menu/Menu.js]: Nearly there man, we just need to be able to set these via props (as an object, i guess), so it's flexible enough to use anywhere

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -47,11 +47,13 @@ const Footer = ({ noSocial, noLinks, campaign, copy }) => {
 
 Footer.propTypes = {
   campaign: propTypes.string.isRequired,
+  fallbackMenu: propTypes.array,
```
Comment (AndyEPhipps) [inline src/components/Footer/Footer.js]: Given that we need this array prop to be a certain structure, I think we're best off defining it properly, something like:

```
   fallbackMenu: PropTypes.arrayOf(PropTypes.shape({
     url: PropTypes.string.isRequired,
     title: PropTypes.string.isRequired,
   })).isRequired,
```

Comment (LouiaR) [reply src/components/Footer/Footer.js]: Done

**Comment by AndyEPhipps** [comment]
I couldn't see this working the CR footer, but noticed the endpoint was being defined elsewhere; have updated it.

Still not sure why Storybook doesn't stop you from not using a REQUIRED prop? When removing the fallBackMenu from the story.js file, there's no error.. which is weird?

**Comment by LouiaR** [comment]
> I couldn't see this working the CR footer, but noticed the endpoint was being defined elsewhere; have updated it.
> 
> Still not sure why Storybook doesn't stop you from not using a REQUIRED prop? When removing the fallBackMenu from the story.js file, there's no error.. which is weird?

Maybe because the default fallback menu is set to an empty array...

**Comment by AndyEPhipps** [comment]
> Maybe because the default fallback menu is set to an empty array...

Ahhh yeah, you're right- we're not supposed to set defaults for a required prop as it doesn't make any sense.

I think we're safe enough to specifically require and set fallbacks within every application we use the footer in; we can always pass an empty array there if we _reeeeally_ don't want it

**Pull request merged**